stubs/openpyxl/openpyxl/workbook/workbook.pyi

```diff
@@ -20,54 +25,54 @@ class Workbook:
-    def remove(self, worksheet) -> None: ...
-    def remove_sheet(self, worksheet) -> None: ...
-    def create_chartsheet(self, title: Incomplete | None = ..., index: Incomplete | None = ...): ...
-    def get_sheet_by_name(self, name): ...
-    def __contains__(self, key): ...
-    def index(self, worksheet): ...
-    def get_index(self, worksheet): ...
-    def __getitem__(self, key): ...
-    def __delitem__(self, key) -> None: ...
-    def __iter__(self): ...
-    def get_sheet_names(self): ...
+    def active(self, value: Worksheet) -> None: ...
+    def create_sheet(self, title: str | None = ..., index: int | None = ...): ...
+    def move_sheet(self, sheet: Worksheet, offset: int = ...) -> None: ...
+    def remove(self, worksheet: Worksheet) -> None: ...
+    def remove_sheet(self, worksheet: Worksheet) -> None: ...
+    def create_chartsheet(self, title: str | None = ..., index: int | None = ...) -> None: ...
+    def get_sheet_by_name(self, name: str) -> Worksheet: ...
+    def __contains__(self, key: str) -> bool: ...
+    def index(self, worksheet: Worksheet) -> int: ...
+    def get_index(self, worksheet: Worksheet) -> int: ...
+    def __getitem__(self, key: str) -> Worksheet: ...
+    def __delitem__(self, key: str) -> None: ...
+    def __iter__(self) -> Iterator[Worksheet]: ...
+    def get_sheet_names(self) -> list[Worksheet]: ...
     @property
-    def worksheets(self): ...
+    def worksheets(self) -> list[Worksheet]: ...
     @property
-    def chartsheets(self): ...
+    def chartsheets(self) -> list[Chartsheet]: ...
     @property
-    def sheetnames(self): ...
+    def sheetnames(self) -> list[str]: ...
     def create_named_range(
-        self, name, worksheet: Incomplete | None = ..., value: Incomplete | None = ..., scope: Incomplete | None = ...
+        self, name: str, worksheet: Worksheet | None = ..., value: str | Any | None = ..., scope: Incomplete | None = ...
     ) -> None: ...
-    def add_named_style(self, style) -> None: ...
+    def add_named_style(self, style: Any) -> None: ...
     @property
-    def named_styles(self): ...
-    def get_named_ranges(self): ...
-    def add_named_range(self, named_range) -> None: ...
-    def get_named_range(self, name): ...
-    def remove_named_range(self, named_range) -> None: ...
+    def named_styles(self) -> list[NamedStyle]: ...
+    def get_named_ranges(self) -> list[Any]: ...
+    def add_named_range(self, named_range: Any) -> None: ...
+    def get_named_range(self, name: Any) -> Any: ...
+    def remove_named_range(self, named_range: Any) -> None: ...
     @property
-    def mime_type(self): ...
-    def save(self, filename) -> None: ...
+    def mime_type(self) -> Any: ...
+    def save(self, filename: str) -> None: ...
     @property
-    def style_names(self): ...
-    def copy_worksheet(self, from_worksheet): ...
+    def style_names(self) -> list[NamedStyle]: ...
+    def copy_worksheet(self, from_worksheet: Worksheet) -> Worksheet: ...
```

Comment:
This returns a `WorksheetCopy` object, not a `Worksheet` object: https://foss.heptapod.net/openpyxl/openpyxl/-/blob/branch/3.0/openpyxl/workbook/workbook.py#L433-435

```suggestion
    def copy_worksheet(self, from_worksheet: Worksheet) -> WorksheetCopy: ...
```

Reply:
It doesn't return WorksheetCopy, but rather a Worksheet or WriteOnlyWorksheet.

The WorksheetCopy operatates on the result of Workbook.create_sheet(), but doesn't itself get returned.

Reply:
Good catch!